cli/determined_cli/checkpoint.py

```diff
@@ -1,13 +1,62 @@
-import sys
+import json
 from argparse import Namespace
+from typing import Any, List
 
-from determined_common import api, storage
+from determined_common import api, constants, experimental, storage
 from determined_common.api import gql
 
 from . import render, user
 from .declarative_argparse import Arg, Cmd
 
 
+def format_validation(validation: gql.validations) -> List[Any]:
+    if not validation:
+        return [None, None]
+
+    if validation.state == constants.COMPLETED:
+        return [constants.COMPLETED, json.dumps(validation.metrics, indent=4)]
+    elif validation.state in (constants.ACTIVE, constants.ERROR):
+        return [validation.state, None]
+    else:
+        raise AssertionError("Invalid validation state: {}".format(validation.state))
+
+
+# TODO(neilc): Report more info about checkpoints and validations.
+def format_checkpoint(checkpoint: gql.checkpoints) -> List[Any]:
+    if not checkpoint:
+        return [None, None]
+
+    if checkpoint.state in (constants.COMPLETED, constants.DELETED):
+        return [checkpoint.state, checkpoint.uuid]
+    elif checkpoint.state in (constants.ACTIVE, constants.ERROR):
+        return [checkpoint.state, None]
+    else:
+        raise AssertionError("Invalid checkpoint state: {}".format(checkpoint.state))
+
+
+def render_checkpoint(checkpoint: experimental.Checkpoint, path: str, quiet: bool = False) -> None:
+    if quiet:
+        print(path)
+        return
+
+    print("Local checkpoint path:")
+    print(path, "\n")
+
+    # Print information about the downloaded step/checkpoint.
+    table = [
+        ["Batch #", checkpoint.batch_number],
+        ["Start Time", render.format_time(checkpoint.start_time)],
+        ["End Time", render.format_time(checkpoint.end_time)],
+        ["Checkpoint UUID", checkpoint.uuid],
+        ["Validation Metrics", json.dumps(checkpoint.validation.metrics, indent=4)],
+    ]
+
+    headers, values = zip(*table)  # type: ignore
+
+    render.tabulate_or_csv(headers, [values], False)
+    print()
```

Comment:
Hmm, not a huge fan of the extra line at the end. Could you put the logic into the loop in `download`?

Reply:
I put it here because you don't want a new line in-between the file paths when you pass `--quiet` to the command. The alternative would be to print in the loop conditionally based on the value of `args.quiet`. Is that what you prefer?

Reply:
Ah, I see. Yeah, I think it would be reasonable to have `render_checkpoint` not take the `quiet` argument and have two separate loops in the caller. Not like the difference in output is a big deal, but I suppose I'm not in the mood to have the user-visible output get sloppy to avoid a minor inconvenience in the code. (And, now that I'm getting into it, it doesn't seem super helpful to pass `quiet` down another level just to immediately exit and print a single attribute of the checkpoint.)